Organization/Organization/MainViewController.m

```diff
@@ -0,0 +1,118 @@
+#import "DetailViewController.h"
+#import "CreateEmployeeViewController.h"
+#import "Employee+CoreDataClass.h"
+#import "Organization+CoreDataClass.h"
+#import "AppDelegate.h"
+#import "Organization-Swift.h"
+
+@implementation MainViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    AppDelegate *delegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+
+    NSFetchRequest *request = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Organization" inManagedObjectContext:delegate.managedObjectContext];
+    [request setEntity:entity];
+    NSArray *organizations = [delegate.managedObjectContext executeFetchRequest:request error:nil];
+    if (organizations.count >= 1)
+    {
+        self.org = organizations[0];
+    }
+    else
+    {
+        self.org = [self primaryInit:delegate.managedObjectContext];
+    }
+}
+
+- (Organization *)primaryInit:(NSManagedObjectContext *)context
+{
+    Organization *organization = [NSEntityDescription insertNewObjectForEntityForName:@"Organization" inManagedObjectContext:context];
+    organization.name = @"Siemens";
+    [[AppDelegate instance] saveContext];
+    return organization;
+}
+
+- (void)didReceiveMemoryWarning
+{
+    [super didReceiveMemoryWarning];
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return self.org.empls.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CellForEmployee"];
+    Employee *object = [self.org.empls objectAtIndex:indexPath.row];
+    cell.textLabel.text = object.fullName;
+    return cell;
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+    if ([segue.identifier isEqualToString:@"showDetail"])
+    {
+        NSIndexPath *indexSelectedRow = [self.tableView indexPathForSelectedRow];
+        UITableViewCell *cell = [self.tableView cellForRowAtIndexPath:indexSelectedRow];
+        NSString *employeeName = cell.textLabel.text;
+        Employee *object = [self.org getEmployeeByFullName:employeeName];
+        DetailViewController *controller = [segue destinationViewController];
+        controller.detailItem = object;
+    }
+
+    if ([segue.identifier isEqualToString:@"AddUser"])
+    {
+        CreateEmployeeViewController *controller = [segue destinationViewController];
+        controller.delegate = self;
+    }
+    
+    if ([segue.identifier isEqualToString:@"editOrg"])
+    {
+        OrganizationInfoViewController *controller = [segue destinationViewController];
+        [controller setOrganization:self.org];
+        [NSNotificationCenter.defaultCenter addObserver:self selector:@selector(tableRandomize) name:@"tableRandomize" object:nil];
```

Comment:
If you ever add an observer to NSNotificationCenter.defaultCenter, you should always remove the observer at some point.